ui/app/src/components/Dialogs/LegacyCodeEditorDialog.tsx

```diff
@@ -192,17 +209,32 @@ export default function LegacyCodeEditorDialog(props: LegacyCodeEditorDialogProp
     dispatch(minimizeDialog({ id }));
   };
 
+  const onClose = () => {
+    if (pendingChanges) {
+      dispatch(
+        showConfirmDialog({
+          title: formatMessage(translations.pendingChanges),
+          onOk: batchActions([closeConfirmDialog(), closeCodeEditorDialog()]),
```

Comment:
We should also be unlocking the content. Ideally, you would ask the form to close so it uses the existing mechanics to clean up (unlock) and report it got closed like when you press cancel button.